pyanaconda/ui/tui/spokes/user.py

```diff
@@ -228,17 +274,22 @@ def apply(self):
             self._user_data.groups.remove("wheel")
 
         # Add or remove the user from userlist as needed
-        if self._create_user and (self._user_data not in self.data.user.userList and self._user_data.name):
-            self.data.user.userList.append(self._user_data)
-        elif (not self._create_user) and (self._user_data in self.data.user.userList):
-            self.data.user.userList.remove(self._user_data)
+        user_list = _get_user_list(self._users_module)
+        if self._create_user and (self._user_data not in user_list and self._user_data.name):
```

Comment:
Don't you want to add __eq__ to UserData to be able check for value, not object identity here?

Reply:
Good point - looks like the current implementation based on how we worked with kickstart data will indeed not work, as each time you get a new user list, you get completely new list of UserData instances. So:

`self._user_data not in user_list`

Will resolve to False every time.

So let's go for an __eq__ implementation. :)

Reply:
Fixed.